File: solutions/next_prime.rb
def next_prime_greater_than(n)
  prime_candidate = n + 1
  while !is_prime?(prime_candidate)
    prime_candidate += 1
  end
  return prime_candidate
end


def is_prime?(n)
  (2..Math.sqrt(n)).to_a.each do |divisor|
    if n % divisor == 0
      return false
    end
  end
  return true
end

start = 1
print "We start with 2 as the first prime number\n"
response = "Y"
while response.downcase[0] == "y"
  start = next_prime_greater_than(start)
  puts start
  print "Do you want the next prime? (Y/N)\n"
  response = gets.strip
end

File: solutions/next_prime.js
function next_prime_gen() {

    var latest_prime = 1;
    function seive(n, base_sieve) {
        var numbers = base_sieve || new Array(n) ;
        if(!base_sieve) {
            for(var i=0; i < numbers.length; i++) {
                numbers[i] = true;
            }
        }else{
            n = 2*base_sieve.length;
            for(var i=base_sieve.length; i < n; i++) {

                numbers[i] = true;
            }
        }

        for(var i=2; i < Math.sqrt(n); i++) {
            if(numbers[i]) {
                for(var j = i*i; j < n; j += i) {
                    numbers[j] = false;
                }
            }
        }
        return numbers;
    }
    var sieve_data = seive(20);

    return function next() {
        var i = latest_prime + 1;
        while(sieve_data[i] != true) {
            if(i >= sieve_data.length-2) {
                sieve_data = seive(i, sieve_data);

            }
            i++;
        }
        latest_prime = i;
        return i;
    }
}

var next = next_prime_gen();
console.log(next());
console.log(next());
console.log(next());
console.log(next());
